go/vt/vtgate/vindexes/vindex.go

```diff
@@ -77,18 +78,18 @@ type SingleColumn interface {
 	// a KeyRange, or a single KeyspaceID.
 	// If the Vindex is non-unique, each id would map to either
 	// a KeyRange, or a list of KeyspaceID.
-	Map(vcursor VCursor, ids []sqltypes.Value) ([]key.Destination, error)
+	Map(ctx context.Context, vcursor VCursor, ids []sqltypes.Value) ([]key.Destination, error)
 
 	// Verify returns true for every id that successfully maps to the
 	// specified keyspace id.
-	Verify(vcursor VCursor, ids []sqltypes.Value, ksids [][]byte) ([]bool, error)
+	Verify(ctx context.Context, vcursor VCursor, ids []sqltypes.Value, ksids [][]byte) ([]bool, error)
```

Comment:
Is it safe to change the `SingleColumn` interface since we expect users to have implemented their custom vindexes too? Even if it is safe, is there a reason why we need it? Is some vindex using the context? Most of the ones I saw were just ignoring it.

Reply:
For this reason I have marked the PR as breaking change.
If any user would have implemented their own vindex (currently not known if anyone has) but this will break their vtgate build and they can fix their fork code of vindex implementation.
There is no change that it will silently allow which would have been risky change.

Secondly, All the lookup vindex that call into Execute method will use it. Basically that marks `true` for `NeedsVCursor()` method.